addons/mail/static/src/core/common/composer.js

```diff
@@ -594,14 +593,8 @@ export class Composer extends Component {
             }
         }
         const attachmentIds = this.props.composer.attachments.map((attachment) => attachment.id);
-        const body = this.props.composer.composerText;
-        const validMentions = this.store.getMentionsFromText(body, {
-            mentionedChannels: this.props.composer.mentionedChannels,
-            mentionedPartners: this.props.composer.mentionedPartners,
-            mentionedRoles: this.props.composer.mentionedRoles,
-        });
```

Comment:
Note: mentions are actually lost on restore (message is not highlighted for target) but it was already a known issue before this PR (we only save the text and not the channels/partners/roles).